bokeh/model.py

```diff
@@ -271,10 +298,10 @@ def __str__(self) -> str:
     __repr__ = __str__
 
     @property
-    def id(self) -> str:
+    def id(self) -> ID:
         return self._id
 
-    name: tp.Union[None, str] = Nullable(String, help="""
+    name: str | None = p.Nullable(p.String, help="""
```

Comment:
What's the rationale for sprinkling `p.` everywhere?

Reply:
AFAIK the only thing that makes use of the "contribute multiple descriptors" functionality is the automatic `_units` properties. If those are cleaned up there's probably no reason the properties themselves couldn't just become the descriptors directly, rather than descriptor factories, and that would be a big simplification.